web-backend/ui/src/main.rs

```diff
@@ -153,6 +151,59 @@ fn valid_repo_url(repo: &str) -> bool {
     true
 }
 
+// Fetches the list of exisiting repositories in analysis database
+async fn get_all_repos() -> Result<Vec<Repo>> {
+    let db = Db::new(None, None, None, None).await.unwrap();
+    let config = Config::new(db);
+    let repos = match config.get_repos().await {
+        Ok(repos) => repos,
+        Err(_error) => {
+            error!("Fetching the list os repositories have failed");
+            vec![]
+        }
+    };
+    Ok(repos)
+}
+
+async fn request_metric_analysis(app: App, repo: String) -> Result<&'static str> {
+    // try to request metrics service
+    let sender = app.metrics_requester.lock().unwrap();
+    if sender
+        .try_send(MetricsRequest::StartAnalysis { repo_url: repo })
+        .is_err()
+    {
+        return Ok("metrics service is busy");
+    }
+    Ok("ok")
+}
```

Comment:
Same here, lets return the error instead of always returning Ok
```suggestion
async fn request_metric_analysis(app: App, repo: String) -> Result<()> {
    // try to request metrics service
    let sender = app.metrics_requester.lock().unwrap();
    sender
        .try_send(MetricsRequest::StartAnalysis { repo_url: repo })?;
      Ok(())       
}
```